File: leetcode/1047.cpp
#include <bits/stdc++.h>
class Solution {
public:
    string removeDuplicates(string s) {
        std::vector<char> stack;
        for(auto& it:s){
            if(stack.empty()){
                stack.push_back(it);
                continue;
            }
            if(stack.back()==it){
                continue;
            }
            stack.push_back(it);
        }
        //std::string result;
        //result=to_string();
        return to_string(stack);
    }
};

File: leetcode/1047.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack=[]
        for rs in s:
            if len(stack)==0:
                stack.append(rs)
                continue
            if rs==stack[-1]:
                stack.pop(-1)
            else:
                stack.append(rs)
        return "".join(stack)